module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Driver/VirtuaTest.php

```diff
@@ -51,4 +70,125 @@ public function setUp(): void
+     * @return Generator
+     */
+    public static function getTestGetMyProfileData(): Generator
+    {
+        yield 'address 2 empty but city set' => [
+            'profiles' => [
+                [
+                    'NAME' => 'Last,First',
+                    'STREET_ADDRESS_1' => 'Address 1',
+                    'STREET_ADDRESS_2' => '',
+                    'POSTAL_CODE' => '01zip',
+                    'TELEPHONE_PRIMARY' => '0-cat-1',
+                    'PATRON_TYPE' => 'test',
+                    'CITY' => 'new City',
+                ],
+            ],
+            'expected' => [
+                'firstname' => 'First',
+                'lastname' => 'Last',
+                'birthdate' => null,
+                'address1' => 'Address 1',
+                'address2' => 'new City',
+                'city' => null,
+                'country' => null,
+                'zip' => '01zip',
+                'phone' => '0-cat-1',
+                'mobile_phone' => null,
+                'expiration_date' => null,
+                'group' => 'test',
+                'home_library' => null,
+            ],
+        ];
+
+        yield 'first name and city empty' => [
+            'profiles' => [
+                [
+                    'NAME' => 'Von Last',
+                    'STREET_ADDRESS_1' => 'Address 1',
+                    'STREET_ADDRESS_2' => 'Address 2',
+                    'POSTAL_CODE' => '01zip',
+                    'TELEPHONE_PRIMARY' => '0-cat-1',
+                    'PATRON_TYPE' => 'test',
+                    'CITY' => '',
+                ],
+            ],
+            'expected' => [
+                'firstname' => '',
+                'lastname' => 'Von Last',
+                'birthdate' => null,
+                'address1' => 'Address 1',
+                'address2' => 'Address 2',
+                'city' => null,
+                'country' => null,
+                'zip' => '01zip',
+                'phone' => '0-cat-1',
+                'mobile_phone' => null,
+                'expiration_date' => null,
+                'group' => 'test',
+                'home_library' => null,
+            ],
+        ];
+    }
+
+    /**
+     * Test get my profile
+     *
+     * @param array $profiles Profiles mocking db select
+     * @param array $expected Expected results
+     *
+     * @return       void
+     * @dataProvider getTestGetMyProfileData
+     */
+    public function testGetMyProfile(array $profiles, array $expected): void
+    {
+        $db = $this->getMockBuilder(Oracle::class)->onlyMethods(['simpleSelect'])
+            ->disableOriginalConstructor()->getMock();
+        $db->expects($this->any())->method('simpleSelect')->willReturn($profiles);
+        // Set dbHandle to false to avoid destructor error
+        $reflection = new ReflectionClass($db);
+        $property = $reflection->getProperty('dbHandle');
+        $property->setAccessible(true);
+        $property->setValue($db, false);
```

Comment:
Would it be easier to just fully mock the connector so you don't have to do all that reflection stuff?

```suggestion
        $db = $this->createMock(Oracle::class);
        $db->expects($this->any())->method('simpleSelect')->willReturn($profiles);
```

Reply:
I tested the suggested method, but it seems like __destruct function will be called even this way. I also tried to create a partial mock, but it did not work as well. So I decided to create a MockObjectTrait which contains helper functions for altering objects

Reply:
We actually already have a [ReflectionTrait](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFindTest/Feature/ReflectionTrait.php#L82-L100) which does the same thing -- you might want to adjust to use that instead!